pkg/client/cache/store.go

```diff
@@ -76,6 +100,53 @@ func (c *cache) Add(obj interface{}) error {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	c.items[key] = obj
+	c.updateIndex(obj)
+	return nil
+}
+
+// updateIndex adds or modifies an object in the index
+// it is intended to be called from a function that already has a lock on the cache
+func (c *cache) updateIndex(obj interface{}) error {
+	if c.indexFunc == nil {
+		return nil
```

Comment:
I could see that.  I guess I could have a default indexFunc that maps to the empty string.